basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
             return constrained_points;
         }
 
+        std::vector<lanelet::BasicPoint2d> create_lanechange_route(int ending_lane_id, double starting_downtrack, double ending_downtrack,
+                                                                   const carma_wm::WorldModelConstPtr &wm,const cav_msgs::VehicleState &state)
+        {
+            std::vector<lanelet::BasicPoint2d> centerline_points = {};
+
+            //Get starting lanelet and ending lanelets
+            auto lanelets = wm->getLaneletsBetween(starting_downtrack, ending_downtrack,true);
+            lanelet::ConstLanelet starting_lanelet =lanelets.front();
+            lanelet::ConstLanelet ending_lanelet = wm->getMap()->laneletLayer.get(ending_lane_id);
+            
+            lanelet::ConstLanelet current_lanelet = starting_lanelet;
+            lanelet::ConstLanelets starting_lane;
+            starting_lane.push_back(starting_lanelet);
+
+            std::vector<lanelet::BasicPoint2d> reference_centerline;
+            bool shared_boundary_found = false;
+            bool is_lanechange_left = false;
+            
+            lanelet::BasicLineString2d current_lanelet_centerline = starting_lanelet.centerline2d().basicLineString();
+            reference_centerline.insert(reference_centerline.end(), current_lanelet_centerline.begin(), current_lanelet_centerline.end());
+
+            while(!shared_boundary_found){
```

Comment:
Is it possible that this infinite loops? If we had a circular track but solid/unpassable boundaries between all left/right lanelets, would we spin here forever? Unlikely to happen that obviously, but graph cycles can often be non-obvious. It may be worth adding some kind of hard upper bound or perhaps a way to track if we've made a loop of some kind.

Reply:
Looking at it further, if we're only looking at lanelets on the route, this may not be an issue.